// ngx-bootstrap-datetime-popup/dist/datetime-popup.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { DatepickerModule, DatepickerConfig } from 'ngx-bootstrap';
import { TimepickerModule, TimepickerConfig } from 'ngx-bootstrap/timepicker';
import { DatetimePopupComponent, OffClickDirective } from './components';
var DatetimePopupModule = /** @class */ (function () {
    function DatetimePopupModule() {
    }
    DatetimePopupModule.forRoot = function () {
        return {
            ngModule: DatetimePopupModule,
            providers: [
                DatepickerConfig,
                TimepickerConfig
            ]
        };
    };
    DatetimePopupModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        FormsModule,
                        DatepickerModule.forRoot(),
                        TimepickerModule.forRoot()
                    ],
                    declarations: [
                        DatetimePopupComponent,
                        OffClickDirective
                    ],
                    exports: [
                        DatetimePopupComponent
                    ],
                    entryComponents: [
                        DatetimePopupComponent
                    ]
                },] },
    ];
    /** @nocollapse */
    DatetimePopupModule.ctorParameters = function () { return []; };
    return DatetimePopupModule;
}());
export { DatetimePopupModule };
